boa/src/value/operations.rs

```diff
@@ -167,14 +167,22 @@ impl Value {
             (Self::Integer(x), Self::Rational(y)) => Self::rational(f64::from(*x).powf(*y)),
             (Self::Rational(x), Self::Integer(y)) => Self::rational(x.powi(*y)),
 
-            (Self::BigInt(ref a), Self::BigInt(ref b)) => Self::bigint(a.as_inner().clone().pow(b)),
+            (Self::BigInt(ref a), Self::BigInt(ref b)) => Self::bigint(
+                a.as_inner()
+                    .clone()
+                    .pow(b)
+                    .map_err(|msg| ctx.construct_range_error(&msg))?,
```

Comment:
```suggestion
                    .map_err(|msg| ctx.construct_range_error(msg))?,
```
Its better if we consume the `String` rather than creating a new string from the `&str`

Reply:
Done